clients/src/main/java/org/apache/kafka/common/TopicIdPartition.java

```diff
@@ -78,6 +78,18 @@ public TopicPartition topicPartition() {
         return topicPartition;
     }
 
+    /**
+     * @return true if topic has same topicId and partition index as topic names some time might be empty.
+    */
+    public boolean same(TopicIdPartition tpId) {
+        if (tpId.topic().isEmpty()) {
+            return topicId.equals(tpId.topicId) &&
+                    topicPartition.partition() == tpId.partition();
+        } else {
+            return this.equals(tpId);
```

Comment:
If topic name is not empty, it means that the topicId is zero. In this case, we just want to match based on topic name, right?

Reply:
Updated this now to be sepcific about topic name exist but id doesnt or other way around